microsoft/testsuites/firewalld/firewalldsuite.py

```diff
@@ -0,0 +1,228 @@
+        "194",
+        "195",
+        "196",
+        "197",
+        "200",
+        "240",
+        "252",
+        "306",
+        "324",
+    ]
+
+    # Aarch64 kernel config has some additional FIB enabled
+    # configs compared to x86_64.
+    # Aarch64 failures list is thus a subset of known failures.
+    if not set(current_failures).issubset(set(known_failures)):
+        raise LisaException(
+            "Found unexpected failures, check logs for details\n"
+            f"expected failure list: {known_failures}\n"
+            f"actual failure list:   {current_failures}\n"
+        )
+
+    # Update the test result data
+    result.set_status(
+        TestStatus.PASSED,
+        f"TOTAL:{total_tests}\nFAILED:{failed_tests}\nSKIPPED:{skipped_tests}\n",
+    )
+
+
+@TestSuiteMetadata(
+    area="firewalld",
+    category="functional",
+    description="""
+    This test suite is to validate firewalld daemon on Azure Linux 3.0.
+    """,
+)
+class FirewalldSuite(TestSuite):
+    def before_case(self, log: Logger, **kwargs: Any) -> None:
+        node: Node = kwargs["node"]
+        if not isinstance(node.os, CBLMariner) or node.os.information.version < "3.0.0":
+            raise SkippedException(
+                UnsupportedDistroException(
+                    node.os, "Firewalld testsuite is supported only on AzureLinux 3.0."
+                )
+            )
+
+    @TestCaseMetadata(
+        description="""
+        This test case runs the firewalld testsuite.
+        The test suite is a collection of tests which run against
+        a local firewalld installation.
+        These tests interacts with both iptables & nftables as backend.
+        The set of tests run are
+        1. cli: firewall-cmd client tests
+        2. dbus: dbus interface tests
+        3. python: python binding tests
+        4. features: firewalld daemon functional tests
+        5. regressions: regression tests
+        The testsuite is provided by the firewalld-test rpm.
+        """,
+        priority=3,
+    )
+    def verify_firewalld(
+        self,
+        node: Node,
+        log: Logger,
+        result: TestResult,
+    ) -> None:
+        # Install the dependencies for running the test suite
+        if isinstance(node.os, CBLMariner):
+            node.os.install_packages(
+                [
+                    "ebtables",
+                    "nftables",
+                    "python3-dbus",
+                    "iproute",
+                    "firewalld",
+                    "firewalld-test",
+                ]
+            )
+
+        # Check if ipv6_rpfilter config is supported
+        # Note: Right now, this is the only known config which
+        # causes failure in starting firewalld daemon.
+        if not _is_ipv6_rpfilter_supported(node):
+            raise SkippedException(
+                "Skipping tests. Needs kernel config CONFIG_FIB_INET &"
+                "CONFIG_FIB_IPV6 enabled to use fib based expressions for"
```

Comment:
"CONFIG_FIB_IPV6 enabled to use fib based expressions for " add one space in the end

Reply:
Done